### PR TITLE
Microsofr linux support

### PR DESCRIPTION
**Pull request by qq624758472**
Microsofr linux support

### CODE REVIEW

**Comment by qwe624758472** [comment]
Why can't I test successfully if I change one line of code?

File: src/basic/virt.c [hunk 1]
```diff
@@ -30,6 +30,7 @@ static const char *const vm_table[_VIRTUALIZATION_MAX] = {
         [VIRTUALIZATION_VMWARE]    = "VMwareVMware",
         /* https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/reference/tlfs */
         [VIRTUALIZATION_MICROSOFT] = "Microsoft Hv",
+        [VIRTUALIZATION_MICROSOFT] = "Micr",
```
Comment (bluca) [inline src/basic/virt.c:33]: Why? Please explain this change

File: src/basic/missing_syscall.h [hunk 2]
```diff
@@ -123,7 +123,7 @@ static inline int missing_setns(int fd, int nstype) {
 /* ======================================================================= */
 
 static inline pid_t raw_getpid(void) {
-#if defined(__alpha__)
+#if defined(__alpha__) || defined(__sw_64__)
```
Comment (bluca) [inline src/basic/missing_syscall.h:126]: If this is a new architecture, adding one ifdef in one place won't be enough. How was this tested?

File: src/basic/virt.c (hunk 1 above)
Comment (qq624758472) [reply src/basic/virt.c:33]: On Microsoft's virtual machine(Hyper-v).  The currently recognized virtual machine type is "vm-other".
But sig.text value in the "detect_vm_cpuid(void)" function is "Micr".   
So "Micr" is the abbreviation of "Microsoft".

File: src/basic/missing_syscall.h (hunk 2 above)
Comment (qq624758472) [reply src/basic/missing_syscall.h:126]: This is just to determine the type of virtual machine.
I still need to continue to study the systemd other source code。

Comment (qq624758472) [reply src/basic/missing_syscall.h:126]: Commit 54a3825c74ed1bdbd72ad07beedaa0778e5b88a2 needs to be revoked, but I don’t know how to do it.

Comment (bluca) [reply src/basic/missing_syscall.h:126]: is there gcc/libc/kernel support for this architecture?

File: src/basic/virt.c (hunk 1 above)
Comment (bluca) [reply src/basic/virt.c:33]: I cannot reproduce this, it works fine here, `sig.text` contains `Microsoft Hv` as expected:

```
$ systemd-detect-virt --version
systemd 247 (247.3-6~bpo10+1)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
$ SYSTEMD_LOG_LEVEL=debug systemd-detect-virt
Failed to read $container of PID 1, ignoring: Permission denied
Found container virtualization none.
No virtualization found in DMI
UML virtualization not found in /proc/cpuinfo.
Virtualization found, CPUID=Microsoft Hv
Found VM virtualization microsoft
microsoft
```

**Review by qwe624758472 — COMMENTED** [review]
This architecture is based on Alpha.
Gcc/libc/kernel, etc. all use alpha architecture code.

File: src/basic/virt.c (hunk 1 above)
Comment (qq624758472) [reply src/basic/virt.c:33]: You are right, it was my mistake that caused the wrong type of virtual machine to be obtained.

File: src/basic/missing_syscall.h (hunk 2 above)
Comment (qq624758472) [reply src/basic/missing_syscall.h:126]: Gcc/libc/kernel, etc. all use alpha architecture code.
sw (申威) based on Alpha architecture.
This is just to determine the type of virtual machine.
I should modify the complete code before submitting.
Thank you for your reply

**Pull request closed without merging**